### ISSUE TITLE
[SmartyPants] Incorrect single quote rendered around Markdown links

### ISSUE THREAD

**Issue by jgarber623** [vmg/redcarpet#624]
First off, thanks for the hard work on Redcarpet!

I've stumbled upon a weird bit of misbehavior related to Markdown links and SmartyPants that I couldn't find documented in any of the existing issues. I'm using Ruby 2.4.1 and Redcarpet 3.4.0 in a Rails 5.1 app. Details are below, but let me know if I can provide any additional examples or background.

## Expected Behavior

I'd expect that a run of Markdown like…

```md
[Jason](https://sixtwothree.org)'s website
```

…when parsed with `Redcarpet::Render::SmartyPants.render` would return:

```md
[Jason](https://sixtwothree.org)&rsquo;s website
```

## Observed Behavior

If the `'` is adjacent to the Markdown link code, an unexpected character entity (`&lsquo;`) is returned:

```
irb(main):011:0> Redcarpet::Render::SmartyPants.render("[Jason](https://sixtwothree.org)'s website")
=> "[Jason](https://sixtwothree.org)&lsquo;s website"
```

If the `'` is moved within the anchor text, the expected character entity (`&rsquo;`) is returned:

```
irb(main):012:0> Redcarpet::Render::SmartyPants.render("[Jason's website](https://sixtwothree.org)")
=> "[Jason&rsquo;s website](https://sixtwothree.org)"
```

For completion's sake, surrounding the Markdown link code with single `'` characters returns:

```
irb(main):013:0> Redcarpet::Render::SmartyPants.render("'[Jason](https://sixtwothree.org)'s website")
=> "&lsquo;[Jason](https://sixtwothree.org)&rsquo;s website"
irb(main):014:0> 
```

**Comment by robin850** [comment]
This has been fixed and will be part of Redcarpet 3.5.0. Thanks for reporting ! :-)

**Comment by jgarber623** [comment]
Thanks, @robin850!
